src/main/java/seedu/address/model/person/Group.java

```diff
@@ -19,13 +19,6 @@ public Group(String name) {
         this.name = name;
     }
 
-    /**
-     * Renames the group to the specified {@code newName}.
-     */
-    public void rename(String newName) {
-        name = newName;
-    }
-
```

Comment:
Since we're removing the `rename` method, can probably make the `name` field in `Group` final. 

Though if I can ask, why did you decide that we should create a new Group with the new name rather than renaming the group? Is it to keep the behaviour consistent with `Person`?

Reply:
Yes, similar to person. Decided against premature optimization and to just make group immutable to gain the benefits of immutability. However, if there is a need to optimize the renaming of groups in future, then we can make it mutable.